src/types.rs

```diff
@@ -348,9 +350,86 @@ impl<T> Line<T>
     }
 }
 
+#[derive(PartialEq, Clone, Debug)]
+pub enum WindingOrder {
+    Clockwise,
+    CounterClockwise,
+}
+
 #[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub struct LineString<T>(pub Vec<Point<T>>) where T: Float;
 
+impl<T: Float> LineString<T> {
+
+    /// Returns the winding order of this line
+    pub fn winding_order(&self) -> WindingOrder {
+        let shoelace = twice_area(self);
+        if shoelace < T::zero() {
+            WindingOrder::Clockwise
+        } else if shoelace > T::zero() {
+            WindingOrder::CounterClockwise
+        } else if shoelace == T::zero() {
+            // what should be done here?
```

Comment:
It looks like `shoelace == 0` when the line is degenerate (empty, single point, multiple coincident points) which brings us back to the validity conversation ( #123 #127 ). 

It will also be zero for linestrings co-incident with the line `y = x` or either of the axes -- (and perhaps some other cases?).

I think changing the return type to `Option<WindingOrder>` would be a reasonable choice here.